"""create case_reject_reason table, insert data

Revision ID: 8acc7f630c70
Revises: 68410d6f3d64
Create Date: 2020-05-19 20:15:33.606480

"""
from alembic import op
import sqlalchemy as sa

# revision identifiers, used by Alembic.
revision = '8acc7f630c70'
down_revision = '68410d6f3d64'
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('case_reject_reason',
                    sa.Column('id', sa.Integer(), nullable=False),
                    sa.Column('name', sa.String(), nullable=True),
                    sa.PrimaryKeyConstraint('id')
                    )
    op.add_column('case_application', sa.Column('reject_reason_id', sa.Integer(), nullable=True))
    op.create_foreign_key('fk_case_application_reject_reason_id', 'case_application', 'case_reject_reason',
                          ['reject_reason_id'], ['id'])

    op.execute(
        '''
        INSERT INTO case_reject_reason (id, name) VALUES (1, 'Altered');
        INSERT INTO case_reject_reason (id, name) VALUES (2, 'Damaged');
        INSERT INTO case_reject_reason (id, name) VALUES (3, 'Duplicate');
        INSERT INTO case_reject_reason (id, name) VALUES (4, 'Outdated');
        INSERT INTO case_reject_reason (id, name) VALUES (5, 'Municipality Incorrect');
        INSERT INTO case_reject_reason (id, name) VALUES (6, 'Signature - Missing');
        INSERT INTO case_reject_reason (id, name) VALUES (7, 'Signature/Name - Illegible');
        INSERT INTO case_reject_reason (id, name) VALUES (8, 'Signature - Wrong Date');
        INSERT INTO case_reject_reason (id, name) VALUES (9, 'Applicant Not Owner');
        INSERT INTO case_reject_reason (id, name) VALUES (10, 'In Queue Dup');
        INSERT INTO case_reject_reason (id, name) VALUES (11, 'Different names');
        INSERT INTO case_reject_reason (id, name) VALUES (12, 'Not signed sufficiently via digital signature');
        INSERT INTO case_reject_reason (id, name) VALUES (13, 'Not authorized to sign');
        INSERT INTO case_reject_reason (id, name) VALUES (14, 'Damaged/Torn application from mail');
        INSERT INTO case_reject_reason (id, name) VALUES (15, 'Other');
        '''
    )
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_constraint('fk_case_application_reject_reason_id', 'case_application', type_='foreignkey')
    op.drop_column('case_application', 'reject_reason_id')
    op.drop_table('case_reject_reason')
    # ### end Alembic commands ###
